plugins/storage.js

```diff
@@ -10,90 +36,54 @@ module.exports = (controller) => {
         },
 
         async setUserDetail(userId, name, value) {
-            // TODO replace with mongo db query
-            const index = details.findIndex(
-                (detail) => detail.userId === userId && detail.name === name
-            );
-            if (index !== -1) {
-                details.splice(index, 1);
-            }
-            details.push({ userId, name, value });
+            return UserDetail.findOneAndUpdate(
+                { userId, name },
+                { value },
+                { new: true, upsert: true }
+            ).exec();
```

Comment:
what does `exec` do?  I've never needed to use that before when using mongoose

Reply:
If you call the method with a callback argument they execute right away. Otherwise they return a Query object which can chain other methods and you need to call exec(). I think some of the model instance commands (like .save()) execute immediately also.